packages/eslint-plugin/src/rules/call-super-on-override.ts

```diff
@@ -0,0 +1,119 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as utils from '../util';
+
+type MessageIds = 'missingSuperMethodCall';
+
+export default utils.createRule<[], MessageIds>({
+  name: 'call-super-on-override',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Require overridden methods to call super.method in their body',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      missingSuperMethodCall:
+        "Use 'super{{property}}{{parameterTuple}}' to avoid missing super class method implementations",
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          topLevel: {
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      'MethodDefinition[override=true][kind="method"]'(
+        node: TSESTree.MethodDefinition,
+      ): void {
+        let methodName = '',
+          methodNameIsLiteral = false,
+          methodNameIsNull = false; // don't add quotes for error message on [null] case
```

Comment:
[Refactor] This code creates a collection of `let` variables once and then never changing their values again. It feels to me like that's an opportunity to  extract into a separate function to make it more clear what's going on with these. Either that, or using a ` = ? ... : ...` ternary inline. What do you think?

Combining with the deduplication suggestion (and again not thinking too hard on the name):

```ts
const [methodName, methodVariant] = determineMethodNameAndVariant(node);
```